generators/generator-base-blueprint.js

```diff
@@ -95,4 +106,17 @@ module.exports = class extends BaseGenerator {
         }
         return useBlueprints;
     }
+
+    migrateBlueprintConfig() {
```

Comment:
I have mixed feelings about this. This will work but the configuration is heavily split now, any custom config in a blueprint will break things for the blueprint. I will strongly suggest to check this with various types of blueprints and validate that it works. 

Meanwhile, we will have #10737 to make sure we are not breaking anything. I feel #10737 is just an additional check but once this is validated we can remove things in the next major version.  WDYT?

Reply:
And also what will happen when there are multiple blueprints?

Reply:
This is a complicated issue, but the real problem of getAllJhipsterConfig is that
`jhipster --blueprints foo,bar`
can be different of
`jhipster --blueprints bar,foo`
The config of the last one will be used.

Reply:
> And also what will happen when there are multiple blueprints?

The first run should settle the config.
If the blueprint want to keep the config separated, it should use this.blueprintConfig

Reply:
Maybe unifying the config using blueprints parameter and config before creating the blueprint can help.

Reply:
This is a tricky problem to solve. That is the reason I want to sort it out before we merge it. For now, AFAICT none of the blueprints fall into this weird case, but we should at least have a clear idea of what we should do.